src/write_gcm.c

```diff
@@ -1093,9 +1082,13 @@ static int wg_oauth2_get_auth_header_nolock(oauth2_ctx_t *ctx,
 static int wg_oauth2_talk_to_server_and_store_result(oauth2_ctx_t *ctx,
     const char *url, const char *body, const char **headers, int num_headers,
     cdtime_t now) {
-  char response[2048];
-  if (wg_curl_get_or_post(response, sizeof(response), url, body,
-      headers, num_headers, 0) != 0) {
+  wg_curl_write_ctx_t write_ctx = {
+     .data = malloc(1),
```

Comment:
`.data = (char *) malloc(sizeof(char))` ? 

Same below

Reply:
I would leave "1". FWIW, sizeof(char) == 1 according to the C standard. PTAL https://drj11.wordpress.com/2007/04/08/sizeofchar-is-1/

Reply:
It is cleaner (IMO) to just set .data to NULL here and elsewhere.  ```realloc``` and ```sfree``` will behave sanely if passed a null pointer. The only thing you need to be careful about is if an allocation *never* happens; i.e. if the request somehow succeeds but the write callback is never called. I doubt that case ever actually happens but if it does, you could handle it on the way out of ```wg_curl_get_or_post```. i.e. if the get or post seems to have succeeded, but .data is still NULL, then malloc a single character *at that time* and populate it with \0 in order to indicate an empty string.
 
Finally, I realize you're using .data=null to indicate "realloc failed" but as I indicated above, that's a bug anyway. So you'll need some other mechanism (perhaps a _Bool field in wg_curl_write_ctx_t) to indicate errors.

Reply:
@kosak is correct -- we are expecting back a string, so let's use a `NULL` to indicate a `realloc` failure and a single-NUL-character string to indicate an empty response. But let's consolidate this logic within `wg_curl_get_or_post`.